### ISSUE TITLE
AddExceptionDescriptionCodeRefactoringProvider error

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#5315]
VS Mac 7.6.0.1937

Clicked at the end of this line:

https://github.com/mono/monodevelop/blob/bd00836e873e1672bb18aac76eed6db81997a8e2/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/RemoteBuildEngine.cs#L199

Command bar was displayed with an error about the AddExceptionDescriptionCodeRefactoringProvider.

IDE log:

```
Roslyn error: Extension_Exception AddExceptionDescriptionCodeRefactoringProvider : Object reference not set to an instance of an object
  at RefactoringEssentials.CSharp.CodeRefactorings.AddExceptionDescriptionCodeRefactoringProvider+<GetActions>d__0.MoveNext () [0x0009d] in /Users/vsts/agent/2.136.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/AddExceptionDescriptionCodeRefactoringProvider.cs:35 
  at RefactoringEssentials.SpecializedCodeRefactoringProvider`1+<ComputeRefactoringsAsync>d__1[T].MoveNext () [0x0024e] in /Users/vsts/agent/2.136.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/Common/SpecializedCodeRefactoringProvider.cs:62 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
ERROR [2018-07-05 14:30:15Z]: Roslyn reported an exception to the user
System.NullReferenceException: Object reference not set to an instance of an object
  at RefactoringEssentials.CSharp.CodeRefactorings.AddExceptionDescriptionCodeRefactoringProvider+<GetActions>d__0.MoveNext () [0x0009d] in /Users/vsts/agent/2.136.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/AddExceptionDescriptionCodeRefactoringProvider.cs:35 
  at RefactoringEssentials.SpecializedCodeRefactoringProvider`1+<ComputeRefactoringsAsync>d__1[T].MoveNext () [0x0024e] in /Users/vsts/agent/2.136.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/Common/SpecializedCodeRefactoringProvider.cs:62 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

> VS bug [#643325](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/643325)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.581 (master)

Pull Request #5354 merged by: Mike Krüger
Author: mono
Commit: 7c38679a7e470d7c286313f8d65632ac8fa2b28f (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.2021 (release-7.6)

Pull Request #5389 merged by: Lluis Sanchez
Author: mono
Commit: 68447b07717901841cb753f256e85c9ee16be76a (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5354 merged by: Mike Kr??ger
Author: mono
Commit: 7c38679a7e470d7c286313f8d65632ac8fa2b28f (xamarin/monodevelop)
